tooling/minimize/src/rvalue.rs

```diff
@@ -235,14 +235,18 @@ impl<'cx, 'tcx> FnCtxt<'cx, 'tcx> {
                         let Type::Ptr(new_ptr_ty) = self.translate_ty_smir(*ty, span) else {
                             rs::span_bug!(span, "ptr to ptr cast to non-pointer");
                         };
+                        let operand = self.translate_operand_smir(operand, span);
                         if old_ptr_ty.meta_kind() == new_ptr_ty.meta_kind() {
-                            let operand = self.translate_operand_smir(operand, span);
                             build::transmute(operand, Type::Ptr(new_ptr_ty))
+                        } else if new_ptr_ty.meta_kind() == PointerMetaKind::None {
+                            build::transmute(
+                                build::get_thin_pointer(operand),
+                                Type::Ptr(new_ptr_ty),
```

Comment:
I think we only allow these casts on raw pointers, i.e. ``new_ptr_ty`` must be `Raw` and therefore the `transmute` here should be unnecessary. Can you make this an assertion?